cli/internal/cacheitem/cacheitem.go

```diff
@@ -31,7 +31,7 @@ type CacheItem struct {
 	tw         *tar.Writer
 	zw         io.WriteCloser
 	fileBuffer *bufio.Writer
-	handle     io.Reader
+	handle     interface{}
```

Comment:
Caches are `Reader | Writer | Closer` but you can't express that in Go. So this goes to `interface{}` and each place that needs it does a type assertion.

#4634 made this partially generic, this finishes the work.